pkg/cli/admin/release/info.go

```diff
@@ -1987,6 +2035,306 @@ func describeBugs(out, errOut io.Writer, diff *ReleaseDiff, dir string, format s
+				fmt.Fprintln(o.Out, elem)
+			}
+		}
+
+		if len(rpmDiff.Changed) > 0 {
+			elements := []string{}
+			for pkg, v := range rpmDiff.Changed {
+				elements = append(elements, fmt.Sprintf("  %s %s → %s", pkg, v.Old, v.New))
+			}
+			writeList("Changed:", elements)
+		}
+		if len(rpmDiff.Removed) > 0 {
+			elements := []string{}
+			for pkg, v := range rpmDiff.Removed {
+				elements = append(elements, fmt.Sprintf("  %s %s", pkg, v))
+			}
+			writeList("Removed:", elements)
+		}
+		if len(rpmDiff.Added) > 0 {
+			elements := []string{}
+			for pkg, v := range rpmDiff.Added {
+				elements = append(elements, fmt.Sprintf("  %s %s", pkg, v))
+			}
+			writeList("Added:", elements)
+		}
+	}
+
+	return nil
+}
+
+type RpmDiff struct {
+	Changed map[string]RpmChangedDiff `json:"changed,omitempty"`
+	Added   map[string]string         `json:"added,omitempty"`
+	Removed map[string]string         `json:"removed,omitempty"`
+}
+
+type RpmChangedDiff struct {
+	Old string `json:"old,omitempty"`
+	New string `json:"new,omitempty"`
+}
+
+func (o *InfoOptions) downloadRpmdb(image imagereference.DockerImageReference, dir string) error {
+	// NOTE: in the future, should check if there's an SBOM and get the rpmlist from that instead
+
+	ref := imagesource.TypedImageReference{Type: imagesource.DestinationRegistry, Ref: image}
+
+	tmpDir := dir + ".work"
+	if err := os.RemoveAll(tmpDir); err != nil {
+		return err
+	}
+	if err := os.MkdirAll(tmpDir, 0755); err != nil {
+		return err
+	}
+
+	// We could be a lot more efficient here by filtering the layers which don't
+	// contain the rpmdb. Though we'd still need to make sure we pick up any user-added
+	// layers in case they added more packages.
+
+	opts := extract.NewExtractOptions(genericiooptions.IOStreams{Out: o.Out, ErrOut: o.ErrOut})
+	opts.SecurityOptions = o.SecurityOptions
+	opts.FilterOptions = o.FilterOptions
+	opts.FileDir = o.FileDir
+	opts.ICSPFile = o.ICSPFile
+	opts.IDMSFile = o.IDMSFile
+	opts.PreservePermissions = false
+
+	opts.Mappings = []extract.Mapping{{
+		ImageRef: ref,
+		// ideally we'd have `From: "usr/share/rpm"` here for the machine-os case, but
+		// the extractor can't handle the rpmdb actually being hardlinks to ostree
+		// objects in the tar stream, which would be filtered out
+		To: tmpDir,
+	}}
+
+	if err := opts.Run(); err != nil {
+		return err
+	}
+
+	var rpmdbPath string
+	rpmdbLocations := []string{"var/lib/rpm", "usr/lib/sysimage/rpm", "usr/share/rpm"}
+	for _, location := range rpmdbLocations {
+		tryPath := filepath.Join(tmpDir, location)
+		if fileinfo, err := os.Lstat(tryPath); os.IsNotExist(err) {
+			continue
+		} else if err != nil {
+			return err
+		} else if fileinfo.IsDir() {
+			rpmdbPath = tryPath
+			break
+		} else {
+			continue
+		}
+	}
+
+	if rpmdbPath == "" {
+		return fmt.Errorf("no rpmdb found in %s", image)
+	}
+
+	if err := os.Rename(rpmdbPath, dir); err != nil {
+		return err
+	}
+
+	if err := os.RemoveAll(tmpDir); err != nil {
```

Comment:
you probably want this in a `defer` block just after creating the temporary directory to avoid leaking on the earlier `return` branches (e.g. `Lstat` failing on something other than non-existance).